lib/tesla/middleware/telemetry.ex

```diff
@@ -13,21 +13,102 @@ if Code.ensure_loaded?(:telemetry) do
 
     end
 
-    :telemetry.attach("my-tesla-telemetry", [:tesla, :request], fn event, time, meta, config ->
+    :telemetry.attach("my-tesla-telemetry", [:tesla, :request, :stop], fn event, measurements, meta, config ->
       # Do something with the event
     end)
     ```
 
+    ## Options
+
+    * `:event_prefix` - a list of atoms to prefix to the telemetry event name. This can be set if you need to distinguish events from different clients. Defaults to `[]`
```

Comment:
I wonder if there would be a way to put this into the `meta` instead (like `client: :my_client` field) so that the event name itself isn't dynamic, but you could still pattern-match on the metadata if you need to.

Reply:
With the metadata filtering, it will be possible to figure out which implementation it was using the hidden property `__module__` in the `env`.